<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1472</width>
    <height>903</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QScrollArea" name="scrollArea">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>1448</width>
         <height>828</height>
        </rect>
       </property>
       <widget class="QGroupBox" name="groupBox">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>341</width>
          <height>841</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="title">
         <string/>
        </property>
        <widget class="QLabel" name="label_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>15</y>
           <width>141</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Image conversion </string>
         </property>
        </widget>
        <widget class="QComboBox" name="conversionChoice">
         <property name="geometry">
          <rect>
           <x>162</x>
           <y>10</y>
           <width>171</width>
           <height>31</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>(None)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>RGB to Bin</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>RGB to Grey Scale</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>RGB to HSV</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>RGB to YCbCr</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>HSV to RGB</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>YCbCr to RGB</string>
          </property>
         </item>
        </widget>
        <widget class="Line" name="line">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>321</width>
           <height>20</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>70</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Histograms</string>
         </property>
        </widget>
        <widget class="QComboBox" name="histChoice">
         <property name="geometry">
          <rect>
           <x>162</x>
           <y>70</y>
           <width>171</width>
           <height>31</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>(None)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>GreyScale Histogram</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>RGB Histogram</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Cumulative Histogram</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Equalize Histogram</string>
          </property>
         </item>
        </widget>
        <widget class="Line" name="line_2">
         <property name="geometry">
          <rect>
           <x>7</x>
           <y>110</y>
           <width>321</width>
           <height>20</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>130</y>
           <width>131</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Transformations </string>
         </property>
        </widget>
        <widget class="QComboBox" name="transChoice">
         <property name="geometry">
          <rect>
           <x>162</x>
           <y>130</y>
           <width>171</width>
           <height>31</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>(None)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Translation</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Rotation</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Scale</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Shear</string>
          </property>
         </item>
        </widget>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>180</y>
           <width>21</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>X</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QLineEdit" name="transX">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>180</y>
           <width>41</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="label_7">
         <property name="geometry">
          <rect>
           <x>90</x>
           <y>180</y>
           <width>21</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QLineEdit" name="transY">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>180</y>
           <width>41</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="label_8">
         <property name="geometry">
          <rect>
           <x>180</x>
           <y>180</y>
           <width>71</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Rotation</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QLineEdit" name="transRotation">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>180</y>
           <width>51</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="label_9">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>230</y>
           <width>51</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Scale</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QLineEdit" name="transScale">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>230</y>
           <width>41</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QPushButton" name="startTrans">
         <property name="geometry">
          <rect>
           <x>162</x>
           <y>230</y>
           <width>171</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
        <widget class="Line" name="line_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>260</y>
           <width>321</width>
           <height>20</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="label_10">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>370</y>
           <width>141</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Contour Detection </string>
         </property>
        </widget>
        <widget class="QComboBox" name="contourChoice">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>370</y>
           <width>171</width>
           <height>31</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>(None)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Naif</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Sobel</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Robert</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Prewitt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Laplacian 4</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Laplacian 8</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>LoG</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>DoG</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Canny</string>
          </property>
         </item>
        </widget>
        <widget class="Line" name="line_4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>400</y>
           <width>321</width>
           <height>20</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="label_11">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>420</y>
           <width>141</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Smoothing</string>
         </property>
        </widget>
        <widget class="QComboBox" name="smooChoice">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>420</y>
           <width>171</width>
           <height>31</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>(None)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Moyen</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Gaussian 4</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Gaussian 8</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Gaussian 3x3</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Gaussian 5x5</string>
          </property>
         </item>
        </widget>
        <widget class="QLabel" name="label_12">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>280</y>
           <width>141</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Threshold</string>
         </property>
        </widget>
        <widget class="QComboBox" name="ThresholdChoice">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>280</y>
           <width>171</width>
           <height>31</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>(None)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Threshold</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Threshold by Hysteresis</string>
          </property>
         </item>
        </widget>
        <widget class="QPushButton" name="startThreshold">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>320</y>
           <width>171</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_23">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>320</y>
           <width>31</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Min</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QLineEdit" name="threMin">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>320</y>
           <width>41</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLineEdit" name="threMax">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>320</y>
           <width>41</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="label_24">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>320</y>
           <width>41</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Max</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="Line" name="line_9">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>350</y>
           <width>321</width>
           <height>20</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="Line" name="line_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>450</y>
           <width>321</width>
           <height>20</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
        <widget class="QComboBox" name="houghChoice">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>470</y>
           <width>171</width>
           <height>31</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>(None)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Hough Lines</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Hough Circles</string>
          </property>
         </item>
        </widget>
        <widget class="QLabel" name="label_13">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>470</y>
           <width>141</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Hough transform</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_25">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>510</y>
           <width>81</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Threshold</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
        <widget class="QLineEdit" name="threHough">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>510</y>
           <width>41</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QPushButton" name="startHough">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>510</y>
           <width>171</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="MinRadiusHough">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>560</y>
           <width>41</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="label_26">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>560</y>
           <width>81</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Min Radius</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
        <widget class="QLabel" name="label_27">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>560</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Max Radius</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
        </widget>
        <widget class="QLineEdit" name="MaxRadiusHough">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>280</x>
           <y>560</y>
           <width>41</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </widget>
       <widget class="QPushButton" name="SaveImg">
        <property name="geometry">
         <rect>
          <x>370</x>
          <y>10</y>
          <width>141</width>
          <height>41</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Save</string>
        </property>
       </widget>
       <widget class="QPushButton" name="RefreshImg">
        <property name="geometry">
         <rect>
          <x>540</x>
          <y>10</y>
          <width>141</width>
          <height>41</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Refresh</string>
        </property>
       </widget>
       <widget class="QPushButton" name="openImg">
        <property name="geometry">
         <rect>
          <x>710</x>
          <y>10</y>
          <width>731</width>
          <height>41</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string>Open Image</string>
        </property>
       </widget>
       <widget class="QLabel" name="label">
        <property name="geometry">
         <rect>
          <x>560</x>
          <y>60</y>
          <width>55</width>
          <height>21</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Code</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="codeImg">
        <property name="geometry">
         <rect>
          <x>350</x>
          <y>90</y>
          <width>541</width>
          <height>751</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_4">
        <property name="geometry">
         <rect>
          <x>1090</x>
          <y>60</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>OpenCV</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="CVImg">
        <property name="geometry">
         <rect>
          <x>900</x>
          <y>90</y>
          <width>541</width>
          <height>751</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1472</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
